drivers/display/display_ili9806e_dsi.c

```diff
@@ -0,0 +1,440 @@
+	const struct ili9806e_config *cfg = dev->config;
+	struct mipi_dsi_device mdev;
+	int ret;
+
+#if !CONFIG_INPUT_GT911
+	if (cfg->reset.port) {
+		if (!gpio_is_ready_dt(&cfg->reset)) {
+			LOG_ERR("Reset GPIO device is not ready!");
+			return -ENODEV;
+		}
+		k_sleep(K_MSEC(1));
+
+		ret = gpio_pin_configure_dt(&cfg->reset, GPIO_OUTPUT_INACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Reset display failed! (%d)", ret);
+			return ret;
+		}
+
+		ret = gpio_pin_set_dt(&cfg->reset, 0);
+		if (ret < 0) {
+			LOG_ERR("Reset display failed! (%d)", ret);
+			return ret;
+		}
+		k_sleep(K_MSEC(1));
+
+		ret = gpio_pin_set_dt(&cfg->reset, 1);
+		if (ret < 0) {
+			LOG_ERR("Enable display failed! (%d)", ret);
+			return ret;
+		}
+		k_sleep(K_MSEC(50));
+	}
+
+#endif
+	/* attach to MIPI-DSI host */
+	if (cfg->pixel_format == PIXEL_FORMAT_RGB_565) {
+		mdev.pixfmt = MIPI_DSI_PIXFMT_RGB565;
+	} else {
+		mdev.pixfmt = MIPI_DSI_PIXFMT_RGB888;
+	}
+	mdev.data_lanes = cfg->data_lanes;
+	mdev.mode_flags = MIPI_DSI_MODE_VIDEO;
+	mdev.timings.hactive = cfg->width;
+	mdev.timings.hbp = ILITEK_ILI9806E_HBP;
+	mdev.timings.hfp = ILITEK_ILI9806E_HFP;
+	mdev.timings.hsync = ILITEK_ILI9806E_HSYNC;
+	mdev.timings.vactive = cfg->height;
+	mdev.timings.vbp = ILITEK_ILI9806E_VBP;
+	mdev.timings.vfp = ILITEK_ILI9806E_VFP;
+	mdev.timings.vsync = ILITEK_ILI9806E_VSYNC;
+
+	ret = mipi_dsi_attach(cfg->mipi_dsi, cfg->channel, &mdev);
+	if (ret < 0) {
+		LOG_ERR("Could not attach to MIPI-DSI host");
+		return ret;
+	}
+
+	if (cfg->backlight.port != NULL) {
+		ret = gpio_pin_configure_dt(&cfg->backlight, GPIO_OUTPUT_ACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Could not configure backlight GPIO (%d)", ret);
+			return ret;
+		}
+	}
+
+	ret = ili9806e_config(dev);
+	if (ret) {
+		LOG_ERR("DSI init sequence failed! (%d)", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+#define ILITEK_ILI9806E_DEFINE(n)                                                                  \
+	static const struct ili9806e_config ili9806e_config_##n = {                                \
+		.mipi_dsi = DEVICE_DT_GET(DT_INST_BUS(n)),                                         \
+		.reset = GPIO_DT_SPEC_INST_GET_OR(n, reset_gpios, {0}),                            \
+		.backlight = GPIO_DT_SPEC_INST_GET_OR(n, bl_gpios, {0}),                           \
+		.data_lanes = DT_INST_PROP_BY_IDX(n, data_lanes, 0),                               \
+		.width = DT_INST_PROP(n, width),                                                   \
+		.height = DT_INST_PROP(n, height),                                                 \
+		.channel = DT_INST_REG_ADDR(n),                                                    \
+		.pixel_format = DT_INST_PROP(n, pixel_format),                                     \
+	};                                                                                         \
+	DEVICE_DT_INST_DEFINE(n, &ili9806e_init, NULL, NULL, &ili9806e_config_##n, POST_KERNEL,    \
+			      CONFIG_APPLICATION_INIT_PRIORITY, &ili9806e_api);
```

Comment:
This is yet another instance of a DSI display driver, where we need the init priority to be lower than the display and MIPI DSI driver.

Perhaps we need to define a second display init priority for parallel mode LCD controllers? It seems like multiple vendors now are using the following display chain:

Parallel LCD controller -> Parallel LCD to DSI video mode adapter -> Display controller

The init priorities need to be like so:
- Parallel LCD controller
- DSI video mode adapter
- DSI display controller driver

I think we should create a custom init priority for the parallel LCD controller drivers to use. These drivers need to start before the DSI video mode adapter, and by default both drivers share an init priority, so either could start first.